gax/retry_policy.h

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_API_CODEGEN_GAPIC_RETRY_POLICY_H_
```

Comment:
The structure for this, based on the existing files in the generator, should be: NAMESPACE1_NAMESPACE2_FILE_NAME_H_

So in this case GOOGLE_GAX_RETRY_POLICY_H_. I'm not sure I particularly like that, but let's be consistent for now.

Reply:
Done.

Reply:
This is the current semantics of the reference implementation in google-cloud-cpp. It does make less sense for movement but does make sense for clone, and clone relies on the copy constructor.
I don't think I have a good idea of what common usage is going to look like. If all clients own and interact with policy objects by unique_ptr, and moving a unique_ptr can't trigger the move constructor of the wrapped object, then neither the move nor the copy constructor are ever used except internally.
The internal google-cloud-cpp Table client has a constructor that takes references to policies as parameters and assigns its policy attributes to clones of the parameters. It doesn't take ownership of the parameters. Is this similar to the workflow we want to use or encourage?